src/components/new-safe/create/steps/StatusStep/useSafeCreationEffects.ts

```diff
@@ -19,36 +19,32 @@ const useSafeCreationEffects = ({
   const chainId = useChainId()
   const [pendingSafe, setPendingSafe] = usePendingSafe()
 
+  // Asynchronously wait for Safe creation
   useEffect(() => {
-    if (status === SafeCreationStatus.SUCCESS) {
-      trackEvent(CREATE_SAFE_EVENTS.CREATED_SAFE)
-
-      // Add the Safe and add names to the address book
-      if (pendingSafe && pendingSafe.safeAddress) {
-        dispatch(
-          updateAddressBook(
-            chainId,
-            pendingSafe.safeAddress,
-            pendingSafe.name,
-            pendingSafe.owners,
-            pendingSafe.threshold,
-          ),
-        )
-      }
-
-      // Asynchronously wait for Safe creation
-      if (pendingSafe?.safeAddress) {
-        pollSafeInfo(chainId, pendingSafe.safeAddress)
-          .then(() => setStatus(SafeCreationStatus.INDEXED))
-          .catch(() => setStatus(SafeCreationStatus.INDEX_FAILED))
-      }
-      return
+    if (pendingSafe?.safeAddress) {
+      pollSafeInfo(chainId, pendingSafe.safeAddress)
+        .then(() => setStatus(SafeCreationStatus.INDEXED))
+        .catch(() => setStatus(SafeCreationStatus.INDEX_FAILED))
     }
+  }, [chainId, pendingSafe?.safeAddress, setStatus])
 
-    if (status === SafeCreationStatus.WALLET_REJECTED) {
-      trackEvent(CREATE_SAFE_EVENTS.REJECT_CREATE_SAFE)
+  // Add Safe to Added Safes and add owner and safe names to Address Book
+  useEffect(() => {
+    if (pendingSafe?.safeAddress && status === SafeCreationStatus.PROCESSING) {
```

Comment:
We are setting `PROCESSING` as soon as the user submits the transaction in their wallet but the transaction could still fail or revert. Should we still add those entries to the address book even if the safe hasn't been deployed?

Reply:
I guess it's a trade-off, I also tend to think it's better to keep SUCCESS like before.

Reply:
Restored SUCCESS in both hooks.